linalg/solvers.hpp

```diff
@@ -154,17 +164,30 @@ class OperatorJacobiSmoother : public Solver
    void Mult(const Vector &x, Vector &y) const;
    void MultTranspose(const Vector &x, Vector &y) const { Mult(x, y); }
 
-   /// Recompute the diagonal using the given Operator.
+   /** @brief Recompute the diagonal using the method AssembleDiagonal of the
+       given new Operator, @a op. */
+   /** Note that (Par)BilinearForm operators are treated similar to the way they
+       are treated in the constructor that takes a BilinearForm parameter.
+       Specifically, this means that the OperatorJacobiSmoother will work with
+       true-dof vectors even though the size of the BilinearForm may be
+       different.
+
+       When the new Operator, @a op, is not a (Par)BilinearForm, any previously
+       set array of essential true-dofs will be thrown away because in this case
+       any essential b.c. will be handled by the AssembleDiagonal method. */
    void SetOperator(const Operator &op);
 
 private:
-   int N;
    Vector dinv;
    const double damping;
-   const Array<int> &ess_tdof_list;
+   const Array<int> *ess_tdof_list; // not owned; may be NULL
```

Comment:
Wouldn't an empty list tell that there are no BC on the rank?

Reply:
Yes, setting the pointer to NULL is the same as having an empty Array. The NULL pointer is used by the new default constructor where we do not have an existing Array to initialize the reference.